File: examples/hello-python/seed.py
# Seed for hello-python: python3 seed.py <db-path> <preset>
import sqlite3
import sys

db_path = sys.argv[1] if len(sys.argv) > 1 else "./hello.db"
preset = sys.argv[2] if len(sys.argv) > 2 else "dev"

presets = {
    "dev": ["snakes have no eyelids", "python 1.0 shipped in 1994", "the GIL is (mostly) gone"],
    "empty": [],
}
if preset not in presets:
    print(f"unknown preset '{preset}'", file=sys.stderr)
    sys.exit(1)

con = sqlite3.connect(db_path)
con.execute("DROP TABLE IF EXISTS facts")
con.execute("CREATE TABLE facts (id INTEGER PRIMARY KEY, fact TEXT NOT NULL)")
con.executemany("INSERT INTO facts (fact) VALUES (?)", [(f,) for f in presets[preset]])
con.commit()
con.close()
print(f"seeded {db_path} with preset '{preset}' ({len(presets[preset])} rows)")

File: examples/hello-web/seed.mjs
// Seed script for hello-web: `node seed.mjs <db-path> <preset>`.
// This is the manifest's `create:` command — backlot resolves {{ns}}/{{preset}}
// and invokes it; the sqlite driver's template capability is just a file copy.
import { DatabaseSync } from 'node:sqlite';

const [dbPath = './hello.db', preset = 'dev'] = process.argv.slice(2);

const db = new DatabaseSync(dbPath);
db.exec('DROP TABLE IF EXISTS greetings');
db.exec('CREATE TABLE greetings (id INTEGER PRIMARY KEY, message TEXT NOT NULL)');

const seeds = {
  dev: ['Hello from the dev preset', 'Deterministic data beats drifted data', 'Bind, don’t rebuild'],
  empty: [],
};

const rows = seeds[preset];
if (!rows) {
  console.error(`unknown preset '${preset}' (have: ${Object.keys(seeds).join(', ')})`);
  process.exit(1);
}
const insert = db.prepare('INSERT INTO greetings (message) VALUES (?)');
for (const m of rows) insert.run(m);
console.log(`seeded ${dbPath} with preset '${preset}' (${rows.length} rows)`);
